src/rpc/blockchain.cpp

```diff
@@ -1232,6 +1243,57 @@ UniValue rollbackchain(const UniValue &params, bool fHelp)
     return NullUniValue;
 }
 
+UniValue reconsidermostworkchain(const UniValue &params, bool fHelp)
+{
+    if (fHelp || params.size() > 1)
+        throw runtime_error("reconsidermostworkchain \"[override]\"\n"
+                            "\nWill rollback the chain if needed and then sync to the most work chain.\n"
+                            "\nArguments:\n"
+                            "1. override      (boolean, optional, default=false)"
+                            "\nResult:\n"
+                            "\nExamples:\n" +
+                            HelpExampleCli("reconsidermostworkchain", "") +
+                            HelpExampleCli("reconsidermostworkchain", "\"true\"") +
+                            HelpExampleRpc("reconsidermostworkchain", "\"true\""));
+
+    // Find pindex of most work chain regardless of whether is is valid or not.
+    LOCK(cs_main);
+
+    // Get the set of chaintips
+    std::set<CBlockIndex *, CompareBlocksByHeight> setTips;
+    setTips = GetChainTips();
+
+    // Find the longest chaintip regardless if it is currently the active one.
+    CBlockIndex *pMostWork = chainActive.Tip();
+    for (CBlockIndex *pTip : setTips)
+    {
+        if (pMostWork->nChainWork < pTip->nChainWork)
+            pMostWork = pTip;
+    }
```

Comment:
If pMostWork == chainActive.Tip() (that is, most work IS your current chain) what happens?  I'd like to either see an explicit test of this case in the python code or see an early return here that proves that the logic results in a no-op.

Reply:
done.  I added a return message that nothing was done